File: lib/Models/product_model.dart
import 'dart:typed_data'; // Asegúrate de importar este paquete

class Product {
  final String productname;
  final String description;
  final double price;
  final Uint8List urlimg; // Cambia el tipo a Uint8List

  Product({
    required this.productname,
    required this.description,
    required this.price,
    required this.urlimg,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      productname: json['productname'],
      description: json['description'],
      price: json['price'].toDouble(),
      urlimg: Uint8List.fromList(
          List<int>.from(json['urlimg']['data'])
      ),
    );
  }
}

File: lib/Server/model/product.model.js
const mongoose = require('mongoose');
const db = require('../config/db');

const { Schema } = mongoose;
const ProductSchema = new Schema({
    productname: {
        type: String,
        required: true,

    },
    description: {
        type: String,
        required: true,

    },
    price: {
        type: Number,
        required: true,
    },
    urlimg: {
        type: Buffer,
        required: true,

    },

});
const ProductModel = db.model('product', ProductSchema);
module.exports = ProductModel;